### ISSUE TITLE
WSS4J2 Wss4jSecurityInterceptor -> validation configuration errors [SWS-961]

### ISSUE THREAD

**Issue by gregturn** [spring-projects/spring-ws#1032]
**[Manuel Dominguez Sarmiento](https://jira.spring.io/secure/ViewProfile.jspa?name=mads1980)** opened **[SWS-961](https://jira.spring.io/browse/SWS-961?redirect=false)** and commented

In our use case, we don't use any validation, just securement. This used to work in previous Spring WS versions that depend on WSS4J 1.x, without setting the validationActions property. However when upgrading to Spring WS 2.3.0 and WSS4J 2.x our XML configuration fails because of the following two issues:

a) NullPointerException in validateMessage() because validationActionsVector is not initialized. This field is only initialized if setValidationActions() is invoked, which should not be necessary if validation is not required. NoSecurity should be configured by default.

b) After configuring validationActions="NoSecurity" our configuration still caused failures (Wss4jSecurityValidationException("No WS-Security header found")), as validation was being attempted anyway. This is due to validationActionsVector being initialized as an empty list by WSSecurityUtil.decodeAction("NoSecurity"). Thus the validationActionsVector.contains(WSConstants.NO_SECURITY) check in validateMessage() fails. An empty vector should be interpreted as no validation required, even if validateRequest or validateResponse are true (which are the defaults)

We resolved this by setting the following properties, which should not be required for our use case:
validationActions="NoSecurity"
validateRequest="false"
validateResponse="false"

---

**Affects:** 2.3.0

**Issue Links:**
- #1018 Setting up a Wss4jSecurityInterceptor as no security still requires WS-Security header
  (_**"is superseded by"**_)

**Backported to:** [2.4.1](https://github.com/spring-projects/spring-ws/milestone/55?closed=1)

3 votes, 9 watchers
